### PR TITLE
Add MSVC support and update CI

### PR DESCRIPTION
**Pull request by kononovk**
Trying build on windows

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: Check push on Ubuntu
```
Comment (MBkkt) [inline .github/workflows/pull_request.yml:1]: - [ ] Bad naming

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+on:
+  push:
```
Comment (MBkkt) [inline .github/workflows/pull_request.yml:4]: - [ ] pull_request

File: .github/workflows/pull_request.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            os: ubuntu-latest,
+            build_type: Debug,
+            cc: "gcc-10", cxx: "g++-10"
+          }
+          # Clang compiler:
+          - {
+            os: ubuntu-latest,
+            build_type: Debug,
+            cc: "clang-12", cxx: "clang++-12"
+          }
```
Comment (MBkkt) [inline .github/workflows/pull_request.yml:27]: Add MSVC last compiler debug build

File: .github/workflows/push.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
+name: Check push on Ubuntu
```
Comment (MBkkt) [inline .github/workflows/push.yml:1]: - [ ] Bad name

File: .github/workflows/push.yml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+          - {
+            os: ubuntu-latest,
+            build_type: Debug,
+            cc: "clang-12", cxx: "clang++-12"
+          }
+          - {
+            os: ubuntu-latest,
+            build_type: RelWithDebInfo,
+            cc: "clang-12", cxx: "clang++-12"
+          }
```
Comment (MBkkt) [inline .github/workflows/push.yml:57]: Add msvc window

**Review by MBkkt — CHANGES_REQUESTED** [review]

File: .github/workflows/pull_request.yml (hunk 3 above)
Comment (kononovk) [reply .github/workflows/pull_request.yml:27]: In next PR

File: .github/workflows/push.yml (hunk 5 above)
Comment (kononovk) [reply .github/workflows/push.yml:57]: In next PR

**Review by MBkkt — APPROVED** [review]

**Pull request merged**